explorer/src/Pos/Explorer/Txp/Local.hs

```diff
@@ -146,68 +121,17 @@ eTxProcessTransactionNoLock itw@(txId, txAux) = reportTipMismatch $ runExceptT $
     -- with data to update. In case of `TxExtra` data is only added, but never updated,
     -- hence `mempty` here.
     let eet = ExplorerExtraTxp mempty hmHistories hmBalances utxoSum
-    let ctx =
-            EProcessTxContext
-            { _eptcExtraBase = eet
-            , _eptcAdoptedBVData = bvd
-            , _eptcUtxoBase = resolved
-            }
-    pRes <-
-        lift $
-        modifyTxpLocalData $
-        processTxDo epoch ctx tipBefore itw mTxTimestamp
-    -- We report 'ToilTipsMismatch' as an error, because usually it
-    -- should't happen. If it happens, it's better to look at logs.
-    case pRes of
-        Left er -> do
-            logDebug $ sformat ("Transaction processing failed: " %build) txId
-            throwError er
-        Right _ ->
-            logDebug
-                (sformat ("Transaction is processed successfully: " %build) txId)
+    pure $
+        EProcessTxContext
+        { _eptcExtraBase = eet
```

Comment:
How about just adding an additional type to `ProcessTxContext`? Something like:
```
data ProcessTxContext a = ProcessTxContext
    { _ptcAdoptedBVData :: !BlockVersionData
    , _ptcUtxoBase      :: !Utxo
    , _ptxAdditional      :: !a
    }
```
Wouldn't that remove the additional `pctx` from `txProcessTransactionAbstract` since you could pass `ext` directly?

Reply:
your suggestion makes sense but I don't get how it helps to remove `ptcx`,
`ext = ExplorerExtra` but `pctx = ExplorerExtraTxp`. so if it doesn't help I don't see advantages, though, it can remove several lines of code.

Reply:
I thought `pctx` was `ProcessTxContext` or `EProcessTxContext`. Ok.

Reply:
or, sorry, you are right, I meant `a=ExplorerExtraTxp` (`a` from your definition of `ProcessTxContext`)

Reply:
Right, but you can have multiple cases since the base structure is the same?
For example:
- in one case `a=ExplorerExtraTxp`
- in another case `a=()` - the _default/base_ structure

Reply:
but why it's better than current approach? it won't reduce the number of type parameters.

Reply:
> Wouldn't that remove the additional `pctx` from `txProcessTransactionAbstract` since you could pass ext directly?

Not on the structure itself, but I thought that it would remove it from the function. Since `a = ext` we don't need another type parameter around. At least, that's what I thought, but I didn't try to compile it.

Reply:
@ksaric it seems you confused `ExplorerExtra` and `ExplorerExtraTxp`.

Reply:
I used your way b/c it reduces number of redundant instances and add opportunity to add new field to `ProcessTxContext` without duplication